crates/chia-consensus/src/gen/messages.rs

```diff
@@ -0,0 +1,271 @@
+use crate::gen::condition_sanitizers::sanitize_hash;
+use crate::gen::sanitize_int::{sanitize_uint, SanitizedUint};
+use crate::gen::validation_error::{first, rest, ErrorCode, ValidationErr};
+use chia_protocol::Bytes32;
+use clvmr::{Allocator, NodePtr};
+use std::sync::Arc;
+
+// these are mode flags used as the first argument to SEND_MESSAGE and
+// RECEIVE_MESSAGE. They indicate which properties of the sender and receiver we
+// commit to, that must match. The mode flags for the sender are shifted left 3 bits.
+pub const PARENT: u8 = 0b100;
+pub const PUZZLE: u8 = 0b010;
+pub const AMOUNT: u8 = 0b001;
+pub const PUZZLEAMOUNT: u8 = 0b011;
+pub const PARENTAMOUNT: u8 = 0b101;
+pub const PARENTPUZZLE: u8 = 0b110;
+pub const COINID: u8 = 0b111;
+
+#[derive(Debug)]
+pub enum SpendId {
+    OwnedCoinId(Arc<Bytes32>),
```

Comment:
it would double the size of `SpendId`. Technically. this *could* be an `Rc`, but pyo3 requires `Send` to pass objects to be run without holding the GIL.
In the tests for create/assert announcement, avoiding the copy was an important step to mitigate DoS attacks. Now that we soft-forked in a limit of 1024, it's still faster to avoid the copy (presumably because of the lower memory footprint.

Reply:
Oh I didn't consider the size of the other variants changing, good point.